test/common/src/main/kotlin/org/codice/compliance/utils/decorators/IdpRedirectResponseDecorator.kt

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.compliance.utils.decorators
+
+import org.apache.cxf.rs.security.saml.sso.SSOConstants.RELAY_STATE
+import org.apache.cxf.rs.security.saml.sso.SSOConstants.SAML_RESPONSE
+import org.apache.cxf.rs.security.saml.sso.SSOConstants.SIGNATURE
+import org.apache.cxf.rs.security.saml.sso.SSOConstants.SIG_ALG
+import org.codice.compliance.saml.plugin.IdpRedirectResponse
+import org.w3c.dom.Node
+
+class IdpRedirectResponseDecorator : IdpRedirectResponse {
```

Comment:
✏️ To convert to a primary ctor and remove the mutable map, you could try this:

```kotlin
class IdpRedirectResponseDecorator// can only instantiate by using extension methods in IdpResponseDecorators.kt
internal constructor(response: IdpRedirectResponse) : IdpRedirectResponse(response) {

    private val paramMap: Map<String, String> by lazy {
        parameters.split("&")
                .map { s -> s.split("=") }
                .associate { s -> s[0] to s[1] }
    }

    init {
        samlResponse = paramMap[SAML_RESPONSE]
        relayState = paramMap[RELAY_STATE]
    }
//...
```

Reply:
Genius

Reply:
I had googled how to turn a stream into a map when reviewing this code, but decided I was spending too much time on it and gave up. Thank you @coyotesqrl 😄